File: gripl/gripl-frontend/src/models/dto/MultiEvaluationRequest.ts

export interface MultiEvaluationRequest {
    models: ModelRunConfig[];
    datasets: number[];
    evaluationDataIds?: number[];
    defaultEvaluationEndpoint: string;
    seed?: number;
    maxConcurrent: number | null;
    repetitions?: number;
    useRag: boolean;
    ragMode: string;
    evaluateRag: boolean;
    useSQLLM: boolean;
    activitiesOnly: boolean;
}

export interface ModelRunConfig {
    label: string;
    evaluationEndpoint?: string | null;
    llmProps?: LlmPropsOverride | null;
}

export interface LlmPropsOverride {
    baseUrl?: string | null;
    modelName?: string | null;
    apiKey?: string | null;
    timeoutSeconds?: number | null;
}

File: gripl/gripl-backend/src/main/kotlin/de/mertendieckmann/griplbackend/model/dto/MultiEvaluationRequest.kt
package de.mertendieckmann.griplbackend.model.dto

import de.mertendieckmann.griplbackend.config.LlmConfig

data class MultiEvaluationRequest(
    val defaultEvaluationEndpoint: String,
    val maxConcurrent: Int = 4,
    val models: List<ModelRunConfig>,
    val seed: Int?,
    val datasets: List<Int>,
    val evaluationDataIds: List<Int> = emptyList(),
    val repetitions: Int = 1,
    val useRag: Boolean = false,
    val ragMode: RagMode = RagMode.HYBRID,
    val evaluateRag: Boolean = true,
    val useSQLLM: Boolean = false,
    val activitiesOnly: Boolean = false,
)

data class ModelRunConfig(
    val label: String,
    val evaluationEndpoint: String? = null,
    val llmProps: LlmConfig.Companion.LlmPropsOverride? = null,
) {
    override fun toString(): String =
        "EvaluationRequest(label=$label, llmProps=${llmProps?.copy(apiKey = llmProps.apiKey?.let { "\"****\"" })})"
}
